Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -15860,4 +15831,90 @@
+        <EffectArray Reference="Weapon,AP_StukovInfestedSiegeTank,Range" Value="3"/>
+    </CUpgrade>
+    <CUpgrade id="AP_InfestedBansheeAdvancedOptics">
+        <AffectedUnitArray value="AP_StukovInfestedBanshee"/>
+    </CUpgrade>
+    <CUpgrade id="AP_InfestedSiegeTankBalancedRoots">
+        <EffectArray Operation="Set" Reference="Weapon,AP_StukovInfestedSiegeTankImpalerTentacle,AllowedMovement" Value="Moving"/>
+    </CUpgrade>
+    <CUpgrade id="AP_InfestedDiamondbackConcentratedSpew">
+        <EffectArray Reference="Weapon,AP_StukovInfestedDiamondBack,Range" Value="2"/>
+        <EffectArray Reference="Weapon,AP_StukovInfestedDiamondBack,MinScanRange" Value="2"/>
+        <EffectArray Reference="Abil,AP_StukovInfestedDiamondbackSnare,AutoCastRange" Value="2"/>
+        <EffectArray Reference="Abil,AP_StukovInfestedDiamondbackSnare,Range" Value="2"/>
+    </CUpgrade>
+    <CUpgrade id="AP_InfestedDiamondbackViolentEnzymes">
+        <EffectArray Reference="Effect,AP_StukovInfestedDiamondBackBileStreamDamage,Amount" Value="8"/>
+    </CUpgrade>
+    <CUpgrade id="AP_InfestedDiamondbackBasicEnsnare">
+        <Icon value="Assets\Textures\btn-ability-zerg-stukov-ensnare.dds"/>
+        <AffectedUnitArray value="AP_StukovInfestedDiamondBack"/>
+    </CUpgrade>
+    <CUpgrade id="AP_ZergExcavatingClaws">
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SpineCrawlerRoot,InfoArray[0].SectionArray[Abils].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SpineCrawlerRoot,InfoArray[0].SectionArray[Actor].DurationArray[Duration]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SpineCrawlerRoot,InfoArray[0].SectionArray[Mover].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SpineCrawlerRoot,InfoArray[0].SectionArray[Stats].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Abils].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Actor].DurationArray[Duration]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Mover].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Stats].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Reference="Unit,AP_SpineCrawlerUprooted,Speed" Value="1"/>
+        <EffectArray Reference="Unit,AP_SporeCrawlerUprooted,Speed" Value="1"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_SpineCrawlerUprooted,SpeedMultiplierCreep" Value="1.4"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_SporeCrawlerUprooted,SpeedMultiplierCreep" Value="1.4"/>

```

Comment:
Infested buildings?

Reply:
added

Reply:
The upgrade specifically targets defensive structures, not all structures.  I think its fine to exclude the production building items, similar to creep stomach.  Envy plans to add creep stomach and this upgrade to infested bunker themselves in a separate pr soon.